power_grid_model_c/power_grid_model/include/power_grid_model/three_phase_tensor.hpp

```diff
@@ -275,7 +280,16 @@ inline bool is_nan(Enum x) {
 
 // is normal
 inline auto is_normal(std::floating_point auto value) { return std::isnormal(value); }
-inline auto is_normal(RealValue<false> const& value) {
+template <std::floating_point T> inline auto is_normal(std::complex<T> const& value) {
+    if (value.real() == T{0}) {
+        return is_normal(value.imag());
+    }
+    if (value.imag() == T{0}) {
+        return is_normal(value.real());
+    }
+    return is_normal(value.real()) && is_normal(value.imag());
```

Comment:
i don't think we have to take care of underflows (subnormal numbers). please let me know if you think that is necessary, then i'll improve this function.